src/main/java/io/logz/jmx2graphite/MetricsPipeline.java

```diff
@@ -31,19 +35,36 @@ public MetricsPipeline(Jmx2GraphiteConfiguration conf, MBeanClient client) {
                                                  conf.getGraphiteProtocol());
         this.client = client;
         this.pollingIntervalSeconds = conf.getMetricsPollingIntervalInSeconds();
+        this.beansWhiteListPattern = conf.getWhiteListPattern();
+        this.beansBlackListPattern = conf.getBlackListPattern();
+
+    }
+
+    public List<MetricBean> getFilteredBeans(List<MetricBean> beans) {
+        List<MetricBean> filteredBeans = beans.stream()
+                .filter(bean -> beansWhiteListPattern.matcher(bean.getName()).find())
+                .collect(Collectors.toList());
+
+        filteredBeans.removeAll(beans.stream()
+                .filter((bean -> beansBlackListPattern.matcher(bean.getName()).find()))
```

Comment:
Why not do the white & black matching against the string, as one filter after another? Save you lots of memory allocation and is more clear / readable